### ISSUE TITLE
Check DA vertexing setup used in `DQM/BeamMonitor/plugins/Vx3DHLTAnalyzer.cc`

### ISSUE THREAD

**Issue by francescobrivio** [cms-sw/cmssw#39339]
Quoting the discussion from https://github.com/cms-sw/cmssw/pull/39285#issuecomment-1239586872:

>```
>%MSG-w DAClusterizerinZ_vect:  PrimaryVertexProducer:pixelVertices  07-Sep-2022 17:23:09 CEST Run: 357900 Event: 3204845
>rejected track t_tkwt 1.67157e-24
>%MSG
>```
>all of these warnings with ludicrous track weights are indicating that the DA is getting in input some tracks with very large IP significance, indications that they are not correctly associated to the right vertex. Perhaps it is worth checking how the DA vertexing is setup for this application.

The DA vertexing setup in `DQM/BeamMonitor/plugins/Vx3DHLTAnalyzer.cc` should be double-checked

FYI @dinardo

**Comment by francescobrivio** [comment]
assign dqm

**Comment by francescobrivio** [comment]
@cms-sw/dqm-l2 this is mainly a reminder for the BeamSpot team (@dinardo) to follow-up on Marco's suggestion.

**Comment by mmusich** [comment]
for starters:

https://github.com/cms-sw/cmssw/blob/435424d670ea3c864b7757527d96b8156b594b60/DQM/Integration/python/clients/beampixel_dqm_sourceclient-live_cfg.py#L100-L101

why these numbers?

**Comment by dinardo** [comment]
Hi Marco,
I guess they are there for historical reasons.
Anyway, I tried to make the fit robust against deviations of the parameters from "nominal" conditions.
If you want we can put them to 0,0

**Comment by mmusich** [comment]
>  Anyway, I tried to make the fit robust against deviations of the parameters from "nominal" conditions.

do we have "proof" of this statement?

**Comment by dinardo** [comment]
> > Anyway, I tried to make the fit robust against deviations of the parameters from "nominal" conditions.
> 
> do we have "proof" of this statement?

What do you mean?  As I said, the code does "move" the initial values to cover possible deviations of the beam spot position with respect to the nominal values. Actually, there is more than just one "movement", I try to move in X, Y, and Z.

> > If you want we can put them to 0,0
> 
> In light of the current position of the Beam Spot (0.17,-0.18) cm, (0.,0.) cm is arguably worse than what we have now.

Sure, we can set the most appropriate values

**Comment by mmusich** [comment]
> As I said, the code does "move" the initial values to cover possible deviations of the beam spot position with respect to the nominal values.

Are the pixel tracking regions adjusted as well?

**Comment by dinardo** [comment]
I don't re-do the tracking. I just fit the vertex "could".
I get what I get as vertices and fit the could with a 3D gaussian.
If the "could" moves with respect to some nominal position, the fit procedure should take that into account by trying different fit starting values to get the fit converging

**Comment by mmusich** [comment]
> I don't re-do the tracking.

right. So the two parameters I pointed above regulate where the pixel tracks are seeded. If for whatever reason those regions are inefficient in selecting hits, tracking (and therefore vertexing) efficiency might be affected in a way dependent on the track kinematics and that cannot be recovered downstream.

**Comment by dinardo** [comment]
Sorry, Marco,
I was thinking of some other parameters.
It's better if I shut up :-)
